tests/demos/test_demos_run.py

```diff
@@ -87,17 +74,33 @@ def py_file(rst_file, tmpdir, monkeypatch):
 
 @pytest.mark.skipcomplex  # Will need to add a seperate case for a complex demo.
 def test_demo_runs(py_file, env):
+    # Add pytest skips for missing imports or packages
+    if basename(py_file) in ("stokes.py", "rayleigh-benard.py", "saddle_point_systems.py", "qg_1layer_wave.py"):
+        if "hypre" not in get_external_packages():
+            pytest.skip("hypre not installed with PETSc")
+
     if basename(py_file) == "qgbasinmodes.py":
         try:
+            # Do not use `pytest.importorskip` to check for slepc4py
```

Comment:
Why not?

Reply:
It isn't sufficient to actually detect whether slepc4py is installed. Both petsc4py and slepc4py require `from xy4py import Xy` to actually load the library. I was having issues on some systems!